globus_automate_client/models.py

```diff
@@ -0,0 +1,392 @@
+    IsString: t.Optional[str]
+    IsBoolean: t.Optional[str]
+    IsTimestamp: t.Optional[str]
+
+    class Config:
+        extra = Extra.forbid
+
+    _enforce_jsonpath = validator(
+        "Variable",
+        "StringEqualsPath",
+        "StringLessThanPath",
+        "StringGreaterThanPath",
+        "StringLessThanEqualsPath",
+        "StringGreaterThanEqualsPath",
+        "NumericEqualsPath",
+        "NumericLessThanPath",
+        "NumericGreaterThanPath",
+        "NumericLessThanEqualsPath",
+        "NumericGreaterThanEqualsPath",
+        "BooleanEqualsPath",
+        "TimestampEqualsPath",
+        "TimestampLessThanPath",
+        "TimestampGreaterThanPath",
+        "TimestampLessThanEqualsPath",
+        "TimestampGreaterThanEqualsPath",
+        allow_reuse=True,
+    )(enforce_jsonpath)
+
+    @root_validator(skip_on_failure=True)
+    def validate_only_one_rule_is_set(cls, values):
+        set_fields = {
+            k
+            for k, v in values.items()
+            if k not in {"Variable", "Next"} and v is not None
+        }
+        if len(set_fields) != 1:
+            raise ValueError(
+                f'A "ChoiceRule" may only contain one expression. Found {", ".join(set_fields)}'
+            )
+        return values
+
+
+class DataTestExpressionWithTransitions(DataTestExpression):
+    Next: str
+
+
+class BooleanExpression(BaseModel):
+    And: t.Optional[t.List[t.Union[DataTestExpression, "BooleanExpression"]]] = Field(
+        None, min_items=1
+    )
+    Or: t.Optional[t.List[t.Union[DataTestExpression, "BooleanExpression"]]] = Field(
+        None, min_items=1
+    )
+    Not: t.Optional[t.Union[DataTestExpression, "BooleanExpression"]] = None
+
+    @root_validator(skip_on_failure=True)
+    def validate_only_one_boolean_choice_rule_set(cls, values):
+        set_fields = {
+            k for k, v in values.items() if k not in {"Next"} and v is not None
+        }
+        if len(set_fields) != 1:
+            raise ValueError(
+                'Exactly one of "And", "Or", or "Not" must be specified in a BooleanExpression'
+            )
+        return values
+
+
+class BooleanExpressionWithTransitions(BooleanExpression):
+    """
+    A Boolean Expression is a JSON object which contains a field named "And",
+    "Or", or "Not". If the field name is "And" or "Or", the value MUST be an
+    non-empty object array of Choice Rules, which MUST NOT contain "Next"
+    fields; the interpreter processes the array elements in order, performing
+    the boolean evaluations in the expected fashion, ceasing array processing
+    when the boolean value has been unambiguously determined.
+
+    The value of a Boolean Expression containing a "Not" field MUST be a single
+    Choice Rule, that MUST NOT contain a "Next" field; it returns the inverse
+    of the boolean to which the Choice Rule evaluates.
+    """
+
+    Next: str
+
+
+class ChoiceState(BaseModel):
+    Type: te.Literal[StateType.Choice]
```

Comment:
I considered that but decided it was a low value add since `Type`is defined explicitly on each state anyways. Getting the `Comment` field via a mixin seemed like unnecessarily complex.